src/main.ts

```diff
@@ -0,0 +1,137 @@
+const trigger_event_name = process.env.GITHUB_EVENT_NAME;
+
+const msteams_webhook_url = core.getInput('webhook_url');
+const job_status = core.getInput('job_status');
+const event_id = core.getInput('event_id');
+
+export const parsedNotificationMessage = async function parse_event_to_message(event_payload_text: string): Promise<NotificationMessage> {
+    let details;
+    let account;
+    let message;
+    let url;
+    let parsedSchema;
+    try {
+        if (job_status.toUpperCase() === "CANCELLED") {
+            console.log("Job was cancelled, no notification will be sent")
+            process.exit(0);
+        }
+
+        let event_name : string = trigger_event_name || "event";
+        if (event_id) {
+            event_name = event_id;
+        }
+
+        switch(event_name){
+            case EventType.PUSH:
+                await schemaOnPush.validate(event_payload_text);
+                parsedSchema = schemaOnPush.cast(event_payload_text);
+                account = parsedSchema.pusher.name;
+                message = `**Commit to GitHub** by ${account}`;
+                url = parsedSchema.compare;
+                details = `Comment: ${parsedSchema.head_commit.message}`;
+                break;
+            case EventType.PULL_REQUEST:
+                await schemaOnPullRequest.validate(event_payload_text);
+                parsedSchema = schemaOnPullRequest.cast(event_payload_text);
+                account = parsedSchema.pull_request.user.login;
+                message =  `**PR submitted to Github** by ${account} with title: ${parsedSchema.pull_request.title}`;
+                url = parsedSchema.pull_request.html_url;
+                details = `Pr for merging ref ${parsedSchema.pull_request.head.ref} to base branch ${parsedSchema.pull_request.base.ref}`;
+                break;
+            case EventType.ISSUE:
+                await schemaOnIssue.validate(event_payload_text);
+                parsedSchema = schemaOnIssue.cast(event_payload_text);
+                account = parsedSchema.sender.login;
+                message = `**New/updated GitHub issue**: ${parsedSchema.issue.title}`;
+                url = parsedSchema.issue.html_url;
+                details = `Issue state: ${parsedSchema.issue.state}  - assignee: ${parsedSchema.issue.assignee}`;
+                break;
+            case EventType.ISSUE_COMMENT:
+                await schemaOnIssueComment.validate(event_payload_text);
+                parsedSchema = schemaOnIssueComment.cast(event_payload_text);
+                account = parsedSchema.sender.login;
+                message = `**A Github issue comment was posted** by ${account} with content ${parsedSchema.comment.body}`;
+                url = parsedSchema.issue.html_url;
+                details = `Issue state: ${parsedSchema.issue.state}  - assignee: ${parsedSchema.issue.assignee}`;
+                break;
+            default:
+                if (event_name) {
```

Comment:
is this really a legitimate default? Or should this be considered an error (genuine question - not implying that it is wrong)

Reply:
it is a legitimate case imho,  I have an example in the readme, where we trigger on push but what we want to comment on actually are the tests completed.
The idea was to make this customizable.